app/gather/assets/apps/survey/SurveyForm.jsx

```diff
@@ -1066,13 +1107,30 @@ class SurveyForm extends Component {
       })
     })
 
-    // The last action is to propagate ODK Project artefacts to Kernel
+    // The action is to propagate ODK Project artefacts to Kernel
     actions.push({
       message: formatMessage(MESSAGES.propagateODKSurvey, { name: this.state.name }),
       method: patchData,
       url: getSurveysAPIPath({ app: ODK_APP, id: this.state.id, action: 'propagate' })
     })
 
+    // The action is to save the survey in gather
+    actions.push({
+      message: formatMessage(MESSAGES.saveGatherSurvey, { name: this.state.name }),
+      method: this.state.gather.project_id ? putData : postData,
+      url: getSurveysAPIPath({ app: GATHER_APP, id: this.state.gather.project_id }),
```

Comment:
```suggestion
      url: getSurveysAPIPath({ app: GATHER_APP, id: this.state.id }),
```
The Gather survey and the kernel project share id.